lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1784,7 +1758,10 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
       this._cacheStorage.setItem(Constants.renewStatus + tokenResponse.stateResponse, Constants.tokenRenewStatusCompleted);
       this._cacheStorage.removeAcquireTokenEntries(authorityKey, acquireTokenUserKey);
       //this is required if navigateToLoginRequestUrl=false
-      this._cacheStorage.clearCookie();
+      if (this.storeAuthStateInCookie) {
+          this._cacheStorage.setItemCookie(authorityKey, '', -1);
```

Comment:
why are we not doing this '`this._cacheStorage.setItemCookie(authorityKey, '', -1);'` inside clearcookie?

Reply:
Because authority key is dynamically created with state unlike state, nonce and loginStartPage. In order to set it inside the clearCookie function, you will have to change the clearCookie api to accept a optional string.